node/src/blockchain/blockchain_bridge.rs

```diff
@@ -874,75 +982,206 @@ mod tests {
+        let blockchain_interface_web3 = make_blockchain_interface_web3(Some(port));
+        let consuming_wallet = make_paying_wallet(b"consuming_wallet");
+        let system = System::new(test_name);
+        let agent = BlockchainAgentMock::default().consuming_wallet_result(consuming_wallet);
+        let msg = OutboundPaymentsInstructions::new(vec![], Box::new(agent), None);
+        let persistent_config = PersistentConfigurationMock::new();
+        let mut subject = BlockchainBridge::new(
+            Box::new(blockchain_interface_web3),
+            Box::new(persistent_config),
+            false,
+        );
+        let (accountant, _, accountant_recording) = make_recorder();
+        subject
+            .pending_payable_confirmation
+            .new_pp_fingerprints_sub_opt = Some(accountant.start().recipient());
+
+        let result = subject
+            .process_payments(msg.agent, msg.affordable_accounts)
+            .wait();
+
+        System::current().stop();
+        system.run();
+        let error_result = result.unwrap_err();
+        assert_eq!(
+            error_result,
+            GasPriceQueryFailed(BlockchainError::QueryFailed(
+                "Decoder error: Error(\"0x prefix is missing\", line: 0, column: 0)".to_string()
+            ))
+        );
+        let recording = accountant_recording.lock().unwrap();
+        assert_eq!(recording.len(), 0);
+    }
+
+    fn assert_sending_error(error: &PayableTransactionError, error_msg: &str) {
+        if let PayableTransactionError::Sending { msg, .. } = error {
+            assert!(
+                msg.contains(error_msg),
+                "Actual Error message: {} does not contain this fragment {}",
+                msg,
+                error_msg
+            );
+        } else {
+            panic!("Received wrong error: {:?}", error);
+        }
+    }
+
+    #[test]
+    fn blockchain_bridge_processes_requests_for_a_complete_and_null_transaction_receipt() {
+        let (accountant, _, accountant_recording_arc) = make_recorder();
+        let accountant = accountant.system_stop_conditions(match_every_type_id!(ScanError));
+        let pending_payable_fingerprint_1 = make_pending_payable_fingerprint();
+        let hash_1 = pending_payable_fingerprint_1.hash;
+        let hash_2 = make_tx_hash(78989);
+        let pending_payable_fingerprint_2 = PendingPayableFingerprint {
+            rowid: 456,
+            timestamp: SystemTime::now(),
+            hash: hash_2,
+            attempt: 3,
+            amount: 4565,
+            process_error: None,
+        };
+        let first_response = ReceiptResponseBuilder::default()
+            .transaction_hash(hash_1)
+            .build();
+        let port = find_free_port();
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .begin_batch()
+            .raw_response(first_response)
+            .raw_response(r#"{ "jsonrpc": "2.0", "id": 1, "result": null }"#.to_string())
```

Comment:
Add a comment above this line, explaining the use of null.
"Null is returned from the RPC server where there is no record of the tx on the blockchain"
Maybe find the description on the RPC docs